canonical-web-and-design/ubuntu.com
# Purchase preview for microcerts

## Done

- Cube preview purchase
- Unified 2 endpoints in 1 by passing parameter `preview=true` if not parameter found it should go to `preview=false`

## QA

- Go to /cube/microcerts and click on purchase
or if you have HTTP client such as Insomnia, POstman, or Curl setup already you can:

POST /cube/microcerts/purchase.json?preview=true&test_backend=true

```
{
  "account_id":"<your account_id>", // This must be your account id on UA contracts
  "value":<price>, // This must be the value provided by GET /cube/microcerts.json
  "product_listing_id":",
  "previous_purchase_id":""
}
```
If successful it should provide data for the modal to do the complete purchase
```
{
  "amountDue": 1200, 
  "currency": "usd", 
  "customerAddress": {
    "city": "London", 
    "country": "GB", 
    "line1": "Bluefin", 
    "line2": "", 
    "postal_code": "12123", 
    "state": ""
  },
  ... 
}
```

## Issue / Card

Fixes #https://github.com/canonical-web-and-design/commercial-squad/issues/262
